lib/jazzy/sourcekitten.rb

```diff
@@ -309,6 +311,35 @@ def self.filter_excluded_files(json)
       end.compact
     end
 
+    def self.names_and_urls(docs)
+      docs.flat_map do |doc|
+        # FIXME: autolink more than just top-level items.
+        [{ name: doc.name, url: doc.url }] # + names_and_urls(doc.children)
+      end
+    end
+
+    def self.autolink_text(text, data, url)
+      regex = /\b(#{data.map { |d| Regexp.escape(d[:name]) }.join('|')})\b/
+      text.gsub(regex) do
+        name = Regexp.last_match(1)
+        auto_url = data.find { |d| d[:name] == name }[:url]
+        if auto_url == url # docs shouldn't autolink to themselves
+          Regexp.last_match(0)
+        else
+          "<a href=\"#{ELIDED_AUTOLINK_TOKEN}#{auto_url}\">#{name}</a>"
+        end
+      end
+    end
+
+    def self.autolink(docs, data)
+      docs.map do |doc|
```

Comment:
I don't think so. The intent is to return an array of transformed elements, not to mutate the elements in-place. So if this is what `map` is doing (passing mutable references to the elements), then making a copy of the doc block argument would be more "pure", but functionally the same I believe.